### DISCUSSION TITLE
Retry and Persistent Storage mechanism

### DISCUSSION THREAD

**Discussion by Anilk549** [ix-ai/smtp#31] [Q&A]
Our application was using IIS SMTP as mail service
We are migrating our applications to kubernetes trying to setup smtp mail service in kubernetes with this image.
There are a couple of problems I'm trying to solve.
1. Does this image supports Retry mechanism? Do we have any configuration support ?
    Let say IIS SMTP has configuration below for retry mechanism.
    
    ![image](https://github.com/ix-ai/smtp/assets/149612144/411fae0c-3fcd-481a-9ca8-5ba69e0cf5a7)

2. Does this image has Persistent storage mechanism? To know the state of the emails?

**Comment by tlex** [comment]
Hello,

Concerning the first question, I don't know what IIS is doing with that setting. I will assume that it tries to connect to the configured service again. So the answer is: it depends on how you configure it. The default setting will not care about it, every connection is treated as a first connection. So ... yes?

For the second question, please see #17. Is you provide storage for `/var/spool/exim4`, yes.

**Reply by xenon-92** [reply]
@tlex looking at the above image I could see that the first retry happens after 15 minutes, 2nd retry after 30 minutes and so on. Can we configure the same settings to this image? I have gone through the documentation but I did not find any.

**Reply by tlex** [reply]
You will need to go through the documentation of [Exim](https://www.exim.org/) to see how to do that. This image is only a packaging for it, with a bit of scripting to help with some (common) configuration items.
